### PR TITLE
New Horizontal scroll support

### PR DESCRIPTION
**Pull request by andirsun**
fixes: #201

### CODE REVIEW

**Comment by jeremypw** [comment]
This works most of the time although I did get into a state where the icon (and popover) stopped responding to scrolling - haven't been able to reproduce that issue yet.  Also had an issue where scrolling on the indicator in the *greeter* caused wingpanel to crash and the login to fail.

**Comment by jeremypw** [comment]
I can confirm that the crash when *horizontal* scrolling in the greeter is reproducible.  I haven't been able to reproduce the login failure thereafter though.

**Comment by jeremypw** [comment]
Found another issue: Very fast random scrolling on the icon (all directions at random) can result in the notification popover no longer showing on any scrolling although the brightness still changes.

**Comment by jeremypw** [comment]
The "handle_scroll_event" code is duplicated from ScreenBrightness.vala - would it be possible to put it in a `src/Utils.vala` and shared?

**Comment by jeremypw** [comment]
Thats weird I've now discovered that the crash in greeter *does* occur in master - but with vertical scrolling not horizontal :shrug:.  I'll raise an issue.

**Comment by jeremypw** [comment]
I've now pushed a PR to fix the crash in greeter.

File: src/Indicator.vala [hunk 1]
```diff
@@ -95,41 +91,59 @@ public class Power.Indicator : Wingpanel.Indicator {
 
         return popover_widget;
     }
+
     /* Smooth scrolling vertical support. Accumulate delta_y until threshold exceeded before actioning */
-    private double total_y_delta= 0;
```
Comment (kvdz) [inline src/Indicator.vala:99]: I think we should move line 96 till line 149 into a separate file. 
for me this clutters the indicator class.

File: .editorconfig [hunk 2]
```diff
@@ -0,0 +1,15 @@
+# EditorConfig <http://EditorConfig.org>
```
Comment (kvdz) [inline .editorconfig:1]: i am not sure if this was intended but it seems not needed to be in the repo.

Comment (andirsun) [reply .editorconfig:1]: Yes, @cassidyjames provide the official file.

File: src/Indicator.vala (hunk 1 above)
Comment (andirsun) [reply src/Indicator.vala:99]: Yep!, Im working on that.

Comment (andirsun) [reply src/Indicator.vala:99]: Done!

**Comment by andirsun** [comment]
> The "handle_scroll_event" code is duplicated from ScreenBrightness.vala - would it be possible to put it in a `src/Utils.vala` and shared? I think this can be left for another PR though as there is some existing duplication in master.

Done!

**Comment by jeremypw** [comment]
@andirsun Great work thank!.  I wonder whether infact the Util handler could also do the brightness change?  It should have access to the DeviceManager service.  There could then be a single correct algorithm to do this and #207 would no longer be required.  Sorry for the mission drift - this could be done in a separate PR if desired.

**Comment by andirsun** [comment]
@jeremypw Done!

**Comment by jeremypw** [comment]
I'd like to get #210 merged first as it addresses a crash then check this PR still works as expected.  If you could test it out on your computer to confirm it works for you I would be grateful. It already has a code approval.

**Comment by andirsun** [comment]
Sure! Let me test it and report.

**Comment by jeremypw** [comment]
@andirsun I have fixed the conflicts with master for you and also noticed a flaw in the logic for showing the notification.  As it was, the notification did not show until the popover had been created and shown.  Now it shows if the popover has never been shown OR it is not currently showing.

**Comment by jeremypw** [comment]
#210 is now merged so this PR can be merged subject to @kvdz satistactory review.

**Comment by jeremypw** [comment]
@kvdz  Are you OK with this PR now?

**Review by jeremypw — APPROVED** [review]
LGTM

**Pull request merged**